Backup fails to continue with `panic: runtime error: slice bounds out of range`

## Output of `restic version`
`restic 0.9.0 compiled with go1.10.2 on linux/amd64`

## How did you run restic exactly?
```
export B2_ACCOUNT_ID=XXX
export B2_ACCOUNT_KEY=XXX
export RESTIC_PASSWORD=XXX
restic --repo b2:XXX:/Data backup /XXX/Data
repository XXX opened successfully, password is correct
panic: runtime error: slice bounds out of range

goroutine 34 [running]:
github.com/restic/restic/internal/ui/termstatus.(*Terminal).SetStatus(0xc42016e5a0, 0xc4204280a0, 0x1, 0x1)
	src/github.com/restic/restic/internal/ui/termstatus/status.go:310 +0x319
github.com/restic/restic/internal/ui.(*Backup).update(0xc42028d450, 0x1, 0x0, 0xd306c00, 0x0, 0x0, 0x0, 0x0, 0xc42058fd00, 0x0)
	src/github.com/restic/restic/internal/ui/backup.go:181 +0x69e
github.com/restic/restic/internal/ui.(*Backup).Run(0xc42028d450, 0xc06fa0, 0xc4200b5200, 0x0, 0x0)
	src/github.com/restic/restic/internal/ui/backup.go:139 +0x5d5
main.runBackup.func2(0x0, 0x0)
	src/github.com/restic/restic/cmd/restic/cmd_backup.go:360 +0x75
github.com/restic/restic/vendor/gopkg.in/tomb%2ev2.(*Tomb).run(0xc4202402d0, 0xc4202383e0)
	src/github.com/restic/restic/vendor/gopkg.in/tomb.v2/tomb.go:163 +0x2b
created by github.com/restic/restic/vendor/gopkg.in/tomb%2ev2.(*Tomb).Go
	src/github.com/restic/restic/vendor/gopkg.in/tomb.v2/tomb.go:159 +0xb9
```

## What backend/server/service did you use to store the repository?
Backblaze

## Expected behavior
Backup continues from where it stopped. Or maybe there could be an error logged to tell me to revert to the previous version of restic to finish the backup.

## Actual behavior
Backup cannot continue.

## Steps to reproduce the behavior
I had a backup running for a week (uploading 350GB to B2 via a crappy ADSL2 line) on restic 0.8.3 and after a reboot I updated to 0.9.0 and tried to continue my backup, unfortunately it now fails with this error.

## Do you have any idea what may have caused this?
I'm guessing that backup that started on 0.8.3 and that have not yet completed cannot be resumed with the newer restic version.


## Do you have an idea how to solve the issue?
Not sure if continuing the backup between version is even thinkable...


## Did restic help you or made you happy in any way?
Really liking the simplicity of restic! It is a great replacement for the convenient-but-bloated CrashPlan home!

--- a/internal/ui/termstatus/status.go
+++ b/internal/ui/termstatus/status.go
@@ -290,6 +290,20 @@ func (t *Terminal) Errorf(msg string, args ...interface{}) {
 	t.Error(s)
 }
 
+// truncate returns a string that has at most maxlen characters. If maxlen is
+// negative, the empty string is returned.
+func truncate(s string, maxlen int) string {
+	if maxlen < 0 {
+		return ""
+	}
+
+	if len(s) < maxlen {
+		return s
+	}
+
+	return s[:maxlen]
+}
+
 // SetStatus updates the status lines.
 func (t *Terminal) SetStatus(lines []string) {
 	if len(lines) == 0 {
@@ -297,7 +311,7 @@ func (t *Terminal) SetStatus(lines []string) {
 	}
 
 	width, _, err := getTermSize(t.fd)
-	if err != nil || width < 0 {
+	if err != nil || width <= 0 {
 		// use 80 columns by default
 		width = 80
 	}
@@ -305,11 +319,7 @@ func (t *Terminal) SetStatus(lines []string) {
 	// make sure that all lines have a line break and are not too long
 	for i, line := range lines {
 		line = strings.TrimRight(line, "\n")
-
-		if len(line) >= width-2 {
-			line = line[:width-2]
-		}
-		line += "\n"
+		line = truncate(line, width-2) + "\n"
 		lines[i] = line
 	}
 

--- a/internal/ui/termstatus/status_test.go
+++ b/internal/ui/termstatus/status_test.go
@@ -1,0 +1,32 @@
+package termstatus
+
+import "testing"
+
+func TestTruncate(t *testing.T) {
+	var tests = []struct {
+		input  string
+		maxlen int
+		output string
+	}{
+		{"", 80, ""},
+		{"", 0, ""},
+		{"", -1, ""},
+		{"foo", 80, "foo"},
+		{"foo", 4, "foo"},
+		{"foo", 3, "foo"},
+		{"foo", 2, "fo"},
+		{"foo", 1, "f"},
+		{"foo", 0, ""},
+		{"foo", -1, ""},
+	}
+
+	for _, test := range tests {
+		t.Run("", func(t *testing.T) {
+			out := truncate(test.input, test.maxlen)
+			if out != test.output {
+				t.Fatalf("wrong output for input %v, maxlen %d: want %q, got %q",
+					test.input, test.maxlen, test.output, out)
+			}
+		})
+	}
+}

